src/client/runner/utils/promise-each.js

```diff
@@ -0,0 +1,23 @@
+import hammerhead from '../deps/hammerhead';
+
+var Promise = hammerhead.Promise;
+
+export default function (items, iterator) {
+    var index = 0;
```

Comment:
I prefer @AndreyBelym's implementation. But don't forget to use our stored version of `reduce`, since `Array.prototype` methods can be overriden